lnrpc/rpc.proto

```diff
@@ -537,6 +537,38 @@ message Utxo {
     int64 confirmations = 6;
 }
 
+enum OutputScriptType {
+    SCRIPT_TYPE_PUBKEY_HASH = 0;
+    SCRIPT_TYPE_SCRIPT_HASH = 1;
+    SCRIPT_TYPE_WITNESS_V0_PUBKEY_HASH = 2;
+    SCRIPT_TYPE_WITNESS_V0_SCRIPT_HASH = 3;
+    SCRIPT_TYPE_PUBKEY = 4;
+    SCRIPT_TYPE_MULTISIG = 5;
+    SCRIPT_TYPE_NULLDATA = 6;
+    SCRIPT_TYPE_NON_STANDARD = 7;
+    SCRIPT_TYPE_WITNESS_UNKNOWN = 8;
+}
+
+message OutputDetail {
+    // The type of the output
+    OutputScriptType output_type = 1;
+
+    // The address
+    string address = 2;
```

Comment:
in the first commit, the OutputDetail struct has an array of addresses instead of just 1. I assumed that is for a multisig script or something? in anycase, if we do a slice there, then this should also be be `repeated`. otherwise, the struct should be changed to just have 1 addr too.

Reply:
The reason to use a slice for it was to deprecate `DestAddresses` but still expose it on the RPC for backwards compatibility. However, we can also re-introduce the `DestAddresses` directly to the interface and have the `OutputDetail` only contain the single destination address if available?

Note that the `DestAddresses` field is a bit weird, because if an output is not re-presentable by an address (e.g. due to `NonStandard` type or the `MultiSigType`), then the indices in the `DestAddresses` do not correspond to the outputs.

The reason that `MultiSigType` returns a slice of addresses, is because this is the case where the multisig script is directly put inside the scriptPubKey, hence not behind a hash. Therefore it cannot be represented by an address. It is questionable if a slice of addresses for the pubkeys referred to in the multisig script is helpful though. :)